README.md

```diff
@@ -0,0 +1,240 @@
+
+If a subscribed registrar wishes to update their billing information,
+they visit their Perma.cc settings page and initiate a request. Just as
+with their original subscription request, a POST of encrypted, non-sensitive
+data is sent to Perma Payments (see views.update), and the user is delivered at
+a page with a self-submitting form that "redirects" them to CyberSource Secure Acceptance Web/Mobile, where they may update their billing information. Perma
+Payments informed of the result of the transaction (see views.cybersource_callback).
+
+
+### Cancelling
+
+CyberSource Secure Acceptance Web/Mobile does not provide a programmatic
+way to cancel a subscription.
+
+Subscribed users may indicate they wish to cancel by visiting their Perma.cc
+settings page. Just as with their original subscription request, a POST of
+encrypted, non-sensitive data is sent to Perma Payments. Perma Payments
+records the request and informs Perma.cc staff. Staff are notified of
+the request immediately, and are additionally sent a daily report of all
+pending cancellation requests, to ensure no requests go astray.
+
+Staff may then cancel the subscription in the CyberSource Business Center,
+following the instructions in the notification email sent by Perma Payments.
+
+
+### Subscription Statuses
+
+In the course of business, Perma.cc needs to know the status of a given
+regsitrar's subscription: do they in fact have a standing subscription?
+Is their payment current? etc.
+
+Perma Payments makes this information available to Perma.cc via a POST-only
+api route (see views.subscription). Using the same communication pattern
+as already described, Perma.cc POSTS a small amount of encrypted, non-sensitive
+data to Perma Payments; Perma Payments verifies the request and POSTs back an
+encrypted response.
+
+#### Note on Status Accuracy
+
+CyberSource does not expose subscription status via an easily-accessible
+API. This has two potential consequences for Perma.cc/Perma Payments:
+
+1. If a customer successfully signs up for a recurring paid subscription
+in CyberSource Secure Acceptance Web/Mobile, and CyberSource's response
+to Perma Payments goes astray, Perma Payments will continue to treat the
+subscription request as 'pending'.
+
+  In the unlikely evident this occurs, a Perma.cc staff member can manually
+  update Perma Payments records when contacted by the affected customer
+  (who will be unable to create links).
+
+2. If a customer with a monthly subscription has credit card trouble on
+a given month and their subscription lapses, Perma Payments will not
+automatically be notified. They will continue to be able to create links.
+
+  To protect against this possibility, Perma.cc staff should periodically
+  update subscription statuses via CSV upload. See "How To" below for
+  detailed instructions.
+
+  Monthly subscriptions are renewed on the first of the month; updating
+  statuses on the 2nd of the month and after any cancellation request
```

Comment:
Can you explain this logic? What does updating a status mean? Is this a customer initiated update or some kind of internal update? Is it updating subscription type, for instance?

If that's the case, then will the user be charged with their old subscription type on the first and then have the updates happen on the 2nd? I doubt this is the case, but the wording is a little confusing

Reply:
I'll change this to something like "Perma.cc staff should periodically retrieve up-to-date information from CyberSource regarding subscription statuses," and will make some other tweaks.

This is just talking about subscription statuses as defined in 80-84, that is, info on whether a registrar has a standing subscription, whether that subscription is paid up or is lapsed, and other info like that. There's a "status" field in the model for storing subscription statuses, and it lists all the possible statuses. CyberSource doesn't make it possible to programmatically check in and see if a customer's subscription is paid up (there's no API route for that). So, if a once-working subscription lapses, due to credit card trouble or similar, Perma Payments would never know. That's not the biggest deal in the world, but it does mean that people would be able to continue making links, for free, as though payment had continued working. To avoid that, Perma.cc staff should periodically ask CyberSource for fresh information about subscription statuses, and update the "status" field in Perma Payments accordingly. That's easy to do using CyberSource's Business Center, and that's what the "how to" instructions walk us through, further down in the doc. This all only affects the subscription "status" field in Perma Payments; no other traits of a subscription, like rate or payment schedule, are affected. Since customers who pay monthly are charged on the 1st of the month, I'm recommending we grab fresh subscription status information on the 2nd of the month.